internal/provider/resource_switch_port_settings.go

```diff
@@ -0,0 +1,1304 @@
+				MTU: func() types.Int32 {
+					if link.Mtu == nil {
+						return types.Int32Null()
+					}
+					return types.Int32Value(int32(*link.Mtu))
+				}(),
+				Speed: types.StringValue(string(link.Speed)),
+			}
+
+			if link.LldpLinkConfig != nil {
+				linkModel.LLDP = &switchPortSettingsLinkLLDPModel{
+					Enabled: types.BoolPointerValue(link.LldpLinkConfig.Enabled),
+				}
+
+				if *link.LldpLinkConfig.Enabled {
+					linkModel.LLDP.ChassisID = func() types.String {
+						if link.LldpLinkConfig.ChassisId == "" {
+							return types.StringNull()
+						}
+						return types.StringValue(link.LldpLinkConfig.ChassisId)
+					}()
+					linkModel.LLDP.LinkDescription = func() types.String {
+						if link.LldpLinkConfig.LinkDescription == "" {
+							return types.StringNull()
+						}
+						return types.StringValue(link.LldpLinkConfig.LinkDescription)
+					}()
+					linkModel.LLDP.LinkName = func() types.String {
+						if link.LldpLinkConfig.LinkName == "" {
+							return types.StringNull()
+						}
+						return types.StringValue(link.LldpLinkConfig.LinkName)
+					}()
+					linkModel.LLDP.ManagementIP = func() types.String {
+						if link.LldpLinkConfig.ManagementIp == "" {
+							return types.StringNull()
+						}
+						return types.StringValue(link.LldpLinkConfig.ManagementIp)
+					}()
+					linkModel.LLDP.SystemDescription = func() types.String {
+						if link.LldpLinkConfig.SystemDescription == "" {
+							return types.StringNull()
+						}
+						return types.StringValue(link.LldpLinkConfig.SystemDescription)
+					}()
+					linkModel.LLDP.SystemName = func() types.String {
+						if link.LldpLinkConfig.SystemName == "" {
+							return types.StringNull()
+						}
+						return types.StringValue(link.LldpLinkConfig.SystemName)
+					}()
+				}
+			}
+
+			if link.TxEqConfig != nil {
+				linkModel.TxEq = &switchPortSettingsLinkTxEqModel{
+					Main: func() types.Int32 {
+						if link.TxEqConfig.Main == nil {
+							return types.Int32Null()
+						}
+						return types.Int32Value(int32(*link.TxEqConfig.Main))
+					}(),
+					Post1: func() types.Int32 {
+						if link.TxEqConfig.Post1 == nil {
+							return types.Int32Null()
+						}
+						return types.Int32Value(int32(*link.TxEqConfig.Post1))
+					}(),
+					Post2: func() types.Int32 {
+						if link.TxEqConfig.Post2 == nil {
+							return types.Int32Null()
+						}
+						return types.Int32Value(int32(*link.TxEqConfig.Post2))
+					}(),
+					Pre1: func() types.Int32 {
+						if link.TxEqConfig.Pre1 == nil {
+							return types.Int32Null()
+						}
+						return types.Int32Value(int32(*link.TxEqConfig.Pre1))
+					}(),
+					Pre2: func() types.Int32 {
+						if link.TxEqConfig.Pre2 == nil {
+							return types.Int32Null()
+						}
+						return types.Int32Value(int32(*link.TxEqConfig.Pre2))
+					}(),
+				}
+			}
+
+			linkModels = append(linkModels, linkModel)
+		}
+
+		model.Links = linkModels
+	}
+
+	//
+	// Routes
+	//
+	if len(settings.Routes) > 0 {
```

Comment:
Same

Reply:
Answered in https://github.com/oxidecomputer/terraform-provider-oxide/pull/426#discussion_r2212336366. Going to consolidate discussion there.